### ISSUE TITLE
d4 file generation (for chanjo2 primarily)

### ISSUE THREAD

**Issue by dnil** [Clinical-Genomics/BALSAMIC#1484]
Compare `raredisease`. To solve https://github.com/Clinical-Genomics/scout/issues/1619 Balsamic could produce d4 files and we could "load" these by giving their filenames in the case load config as `d4_path` (e.g. https://clinical-genomics.github.io/scout/admin-guide/load-config/). This enables `chanjo2` coverage report generation for the case.

**Comment by northwestwitch** [comment]
Thanks for opening this one @dnil!

**Comment by northwestwitch** [comment]
> Hey! I have this PR now that adds the d4file [#1592](https://github.com/Clinical-Genomics/BALSAMIC/pull/1592) would it work directly you think if I just add the file to the case load config? I thought I might give it a try and see how this chanjo2 report might look for a case.

Nice! It should work I think!

**Comment by mathiasbio** [comment]
Yeah it seems to work actually. See here: https://scout-stage.scilifelab.se/cust087/F0043044-2

However, I don't know exactly if I should generate the d4 with some specific list of genes or something. Though I can see that it's possible to search for the completeness of a gene in Chanjo2, I don't know if it counts both introns and exons? Let me know if you know how people usually use chanjo :D

**Comment by northwestwitch** [comment]
> Yeah it seems to work actually. See here: https://scout-stage.scilifelab.se/cust087/F0043044-2

Cool, but so little coverage (5x), is it intended?

> However, I don't know exactly if I should generate the d4 with some specific list of genes or something. 

It works like this: if your case is loaded with gene panels (in case config file) then these are the genes that are used to create the coverage report in chanjo2. I would leave it as it is if unsure

> Though I can see that it's possible to search for the completeness of a gene in Chanjo2, I don't know if it counts both introns and exons? 

This one works like this: if at least one of your samples has WGS, then checks both introns and exons, otherwise if you have WES sata it checks onlt the exons.

Let me know if you know how people usually use chanjo :D

I think they are interested on the coverage over the chosen panels, mostly. Then if they have a variant hitting a gene then I guess check the coverage over the gene and the transcripts specifically, I guess ..

**Comment by mathiasbio** [comment]
Okay! Then probably chanjo2 is interpreting this sample as WGS and that is why the coverage is low, because while this is a panel we do not give any list of genes or bedfile to the case load scout yaml. So probably in this case then, when I search for a gene it is looking at both introns and exons? 🤔  Do you have any ideas on how this cold be changed? I see in one of the MIP scout yamls a gene panel section like this:

```
gene_panels:
- BRAIN
- Cardiology
- CILM
- CH
- CHILDHYPOTONIA
- CTD
- DIAB
- ENDO
- EP
- HEARING
- HYDRO
- IBMFS
- IEM
- IF
- mcarta
- MHT
- MIT
- MOVE
- mtDNA
- NBS-M
- NEURODEG
- NMD
- OMIM-AUTO
- OPTIC
- PANELAPP-GREEN
- PEDHEP
- PID
- PIDCAD
- RETINA
- SKD
- SOVM
- STROKE
- AID
- Inherited cancer
- VEO-IBD
- Kidney
- CAKUT
- ncRNA
```
I guess this must refer to a genelist within Scout. Maybe we can find some genelist like that for our customers. I know for instance that cust087 has some gene panels like this:

<img width="389" height="266" alt="Image" src="https://github.com/user-attachments/assets/7b0d5868-0e48-4f5d-95b5-d5f115862d97" />

Maybe those could be used

**Comment by northwestwitch** [comment]
> Okay! Then probably chanjo2 is interpreting this sample as WGS and that is why the coverage is low, because while this is a panel we do not give any list of genes or bedfile to the case load scout yaml.

Yes, probably!

>Do you have any ideas on how this cold be changed?

I guess there's actually nothing wrong with this behaviour, since it must be true that the coverage is so low considering the whole genome (not only exons + introns), but the whole thing (this is the behavior in chanjo2 -> https://github.com/Clinical-Genomics/chanjo2/pull/347).

 I see in one of the MIP scout yamls a gene panel section like this:
> 
> ```
> gene_panels:
> - BRAIN
> - Cardiology
> - CILM
> - CH
> - CHILDHYPOTONIA
> - CTD
> - DIAB
> - ENDO
> - EP
> - HEARING
> - HYDRO
> - IBMFS
> - IEM
> - IF
> - mcarta
> - MHT
> - MIT
> - MOVE
> - mtDNA
> - NBS-M
> - NEURODEG
> - NMD
> - OMIM-AUTO
> - OPTIC
> - PANELAPP-GREEN
> - PEDHEP
> - PID
> - PIDCAD
> - RETINA
> - SKD
> - SOVM
> - STROKE
> - AID
> - Inherited cancer
> - VEO-IBD
> - Kidney
> - CAKUT
> - ncRNA
> ```
> 
> I guess this must refer to a genelist within Scout. Maybe we can find some genelist like that for our customers. I know for instance that cust087 has some gene panels like this:

Yes, it's the panels already available in Scout!

> 
> <img alt="Image" width="389" height="266" src="https://private-user-images.githubusercontent.com/14308912/470288123-7b0d5868-0e48-4f5d-95b5-d5f115862d97.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTMzNjAwMzEsIm5iZiI6MTc1MzM1OTczMSwicGF0aCI6Ii8xNDMwODkxMi80NzAyODgxMjMtN2IwZDU4NjgtMGU0OC00ZjVkLTk1YjUtZDVmMTE1ODYyZDk3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzI0VDEyMjIxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI0MzUwOGZmMzc1NDZhOTdiOTUzOWZlODdhNjlmNDZjNmY1NDJmZGE2YWNlMGNjNDFjOWMwMTExNmQ2YjAzYjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.3j0uTVQEMGy2W16r1TQybOPLDGUjAWF0b9tun9kYIec">
> Maybe those could be used

I guess it would be a nice idea, otherwise that report is not so informative if it only contains that low coverage without any gene reference. On the other hand it might be just fine if the user is aware of this thing and is using the search form to look at the situation on specific genes..

**Comment by northwestwitch** [comment]
> but if there was a way to "activate" the panel mode in chanjo2 without needing to actually supply a gene panel, then I think that could be a nice solution!

At the moment there isn't, but if you think it would help to treat "panel" sequencing data as WES, hence compute coverage stats over the exons instead, this could be done quite easily in scout?
